src/js/component.js

```diff
@@ -100,38 +102,10 @@ class Component {
     this.childIndex_ = {};
     this.childNameIndex_ = {};
 
-    let SetSham;
-
-    if (!window.Set) {
```

Comment:
Creating a class on every component creation probably isn't idea, even if we don't go with this pull request we should really consider moving this to it's own file.